pkgs/by-name/sh/shiori/package.nix

```diff
@@ -15,6 +15,12 @@ buildGoModule rec {
     sha256 = "sha256-gMIpDiA5ncZ50WZ2Y57mScTEXzeObgZxP+nkWe+a8Eo=";
   };
 
+  prePatch = ''
+    substituteInPlace main.go \
+      --replace-fail 'version = "dev"' 'version = "${version}"' \
+      --replace-fail 'commit  = "none"' 'commit  = "tag v${version}"'
+  '';
```

Comment:
You can set these with ldflags: https://github.com/go-shiori/shiori/blob/660d425288e5d47b0f0cdcd03c6fa77cece11f6b/Makefile#L14

Reply:
Much cleaner indeed, done!